### ISSUE TITLE
Brackets trigger syntax error but are missed by shellcheck

### ISSUE THREAD

**Issue by roaima** [koalaman/shellcheck#2671]
#### For bugs
- Rule Id (if any, e.g. SC1000): No rule triggered
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash                                                                                                                                                                                                   
i=4
echo '==' fitInvMass.C+("$i")
```

#### Here's what shellcheck currently says:
```
No issues detected!
```
#### Here's what I wanted or expected to see:

When `bash` executes this script I get this error,
```
myscript: line 3: syntax error near unexpected token `('
myscript: line 3: `echo '==' fitInvMass.C+("$i")'
```

The `echo '=='` is a placeholder for a command that a user was [trying to include](https://unix.stackexchange.com/q/732883/100397).

I would expect shellcheck to spot the syntax error, and maybe suggest that the string containing the `(…)` brackets should be double-quoted. (The original expression was `fitInvMass.C+($i)` and shellcheck suggested "_[SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting_" for the `$i` variable.)

**Comment by kusalananda** [comment]
This is likely due to accepting `+("$i")` as an extended globbing pattern. However, ShellCheck accepts the syntax even though the `extglob` shell option is not enabled.

I recognize that in the most general case, ShellCheck would likely have to evaluate the given shell code to figure out what shell options are set at any particular part of a script if it wants to find issues like these and that this may be too much to ask. But some form of heuristics could maybe be considered, such as an option not being mentioned anywhere in the code?
